src/tokenManagement/tokenManagementTransactionsOutcomeParser.ts

```diff
@@ -349,6 +359,135 @@ export class TokenManagementTransactionsOutcomeParser {
         };
     }
 
+    parseModifyRoyalties(transaction: TransactionOnNetwork): ModifyRoyaltiesOutput[] {
+        this.ensureNoError(transaction.logs.events);
+
+        const events = findEventsByIdentifier(transaction, "ESDTModifyRoyalties");
+        return events.map((event) => this.getOutputForESDTModifyRoyaltiesEvent(event));
+    }
+
+    private getOutputForESDTModifyRoyaltiesEvent(event: TransactionEvent): ModifyRoyaltiesOutput {
+        const tokenIdentifier = this.extractTokenIdentifier(event);
+        const nonce = this.extractNonce(event);
+        const royalties = !event.topics[3]?.length
+            ? BigInt(0)
+            : BigInt(new BigNumber(Buffer.from(event.topics[3]).toString("hex"), 16).toFixed());
+
+        return {
+            tokenIdentifier: tokenIdentifier,
+            nonce: nonce,
+            royalties,
+        };
+    }
+
+    parseSetNewUris(transaction: TransactionOnNetwork): SetNewUrisOutput[] {
+        this.ensureNoError(transaction.logs.events);
+
+        const events = findEventsByIdentifier(transaction, "ESDTSetNewURIs");
+        return events.map((event) => this.getOutputForESDTSetNewURIsEvent(event));
+    }
+
+    private getOutputForESDTSetNewURIsEvent(event: TransactionEvent): SetNewUrisOutput {
+        const tokenIdentifier = this.extractTokenIdentifier(event);
+        const nonce = this.extractNonce(event);
+        const uri = !event.topics[3]?.length ? "" : event.topics[3].toString();
```

Comment:
I find it a bit strange to have `not  (!)` before the condition :) Not big deal nonetheless.